### PR TITLE
feat: add external ConfigMap-based runtime configuration

### PR DESCRIPTION
**Pull request by tremes**
Add proposal, design, specs, and tasks for making the adapter's operational parameters (pollInterval, initialDelay, cooldownWindow) configurable at runtime via a ConfigMap, without requiring Pod restarts.

Assisted-by: Claude Code:claude-opus-4-6

## Summary by CodeRabbit

* **New Features**
  * Adapter now reads runtime settings from the `alerts-adapter-config` ConfigMap on every poll cycle (`pollInterval`, `initialDelay`, `cooldownWindow`), applying updates on the next tick with safe fallbacks.
  * Configurable Proposal skills/tooling can be set via `tools.skills` and per-step overrides in the same ConfigMap.
* **Documentation**
  * Added/updated specs and design notes for per-cycle config loading, defaults, and fallback/logging behavior.
* **Tests**
  * Added unit tests for ConfigMap loading, duration validation (including non-positive values), and skills parsing/building.
* **Chores**
  * Added the ConfigMap and `POD_NAMESPACE` to the Deployment; updated RBAC/manifests to align required permissions.

### CODE REVIEW

File: openspec/changes/external-configmap-config/tasks.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- [ ] 1.3 Implement `ConfigMapConfigSource` that reads the `alerts-adapter-config` ConfigMap via controller-runtime `client.Get`, parses the `config.yaml` key, and falls back to defaults on NotFound, missing key, invalid YAML, or invalid duration values
+- [ ] 1.4 Add tests for YAML parsing: valid full config, partial config, invalid durations, invalid YAML, empty data, missing key
+- [ ] 1.5 Add tests for `ConfigMapConfigSource.Load`: ConfigMap exists, ConfigMap not found, ConfigMap without config.yaml key
+
+## 2. Adapter integration
+
+- [ ] 2.1 Define `ConfigSource` interface in `internal/adapter` with `Load(ctx context.Context) config.Config`
+- [ ] 2.2 Add `ConfigSource` field to `Adapter` struct and accept it in `New()`
+- [ ] 2.3 Update `reconcile()` to call `ConfigSource.Load()` at the start and use the returned values for initialDelay and cooldownWindow
+- [ ] 2.4 Update `Run()` to detect pollInterval changes and reset the ticker
```
[the remark this answers is not in the document]
Comment (tremes) [reply openspec/changes/external-configmap-config/tasks.md]: Is this resolved?

**Comment by tremes** [comment]
@coderabbit review

File: openspec/changes/external-configmap-config/design.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+The YAML is unmarshalled into a struct with `yaml.Unmarshal`. Duration fields use Go duration string format (`30s`, `5m`, `1h`). A custom `Duration` type wraps `time.Duration` with YAML unmarshalling support.
+
+| Field | Type | Default | Example |
+|-------|------|---------|---------|
+| `pollInterval` | duration string | `30s` | `45s`, `1m` |
+| `initialDelay` | duration string | `5m` | `10m`, `2m30s` |
+| `cooldownWindow` | duration string | `1h` | `30m`, `2h` |
+
+All fields are optional. Missing fields use the default value. Invalid duration strings fall back to the default for that field with a warning log.
```
[the remark this answers is not in the document]
Comment (tremes) [reply openspec/changes/external-configmap-config/design.md:70]: Is this still valid?

File: openspec/changes/configurable-toolspec/specs/tools-config/spec.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- **THEN** the loaded `Config` SHALL contain the corresponding `Skills` entries with their images and paths
+
+#### Scenario: ConfigMap has no skills key
+- **WHEN** the ConfigMap `config.yaml` does not contain a `skills` key
+- **THEN** the loaded `Config` SHALL have an empty `Skills` slice and no error is logged
+
+#### Scenario: ConfigMap does not exist
+- **WHEN** the ConfigMap is not found in the cluster
+- **THEN** the loaded `Config` SHALL use defaults for all timing parameters and have an empty `Skills` slice
+
```
[the remark this answers is not in the document]
Comment (tremes) [reply openspec/changes/configurable-toolspec/specs/tools-config/spec.md]: I think this is fine for now.

**Comment by tremes** [comment]
@coderabbit full review

File: manifests/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  name: lightspeed-agentic-alerts-adapter-proposals
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: lightspeed-agentic-alerts-adapter-proposals
+subjects:
+  - kind: ServiceAccount
+    name: lightspeed-agentic-alerts-adapter
+    namespace: openshift-lightspeed
+---
```
[the remark this answers is not in the document]
Comment (tremes) [reply manifests/rbac.yaml:23]: Is this correct? Proposals are listed cluster wide

File: openspec/changes/external-configmap-config/specs/configmap-config/spec.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+- **WHEN** the `alerts-adapter-config` ConfigMap exists and its `config.yaml` key contains valid YAML with valid duration values
+- **THEN** the system uses the values from the ConfigMap for that reconcile cycle
+
+#### Scenario: ConfigMap exists with partial YAML
+- **WHEN** the `alerts-adapter-config` ConfigMap exists and its `config.yaml` key contains valid YAML with only some fields specified
+- **THEN** the system uses the specified values and falls back to defaults for missing fields
+
+#### Scenario: ConfigMap exists with invalid duration value
+- **WHEN** the `alerts-adapter-config` ConfigMap exists and its `config.yaml` key contains an unparseable duration string for one or more fields
+- **THEN** the system falls back to the default value for each invalid field and logs a warning for each invalid field
```
[the remark this answers is not in the document]
Comment (tremes) [reply openspec/changes/external-configmap-config/specs/configmap-config/spec.md:16]: This is fixed now.

File: openspec/changes/external-configmap-config/tasks.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- [x] 1.3 Implement `ConfigMapConfigSource` that reads the `alerts-adapter-config` ConfigMap via controller-runtime `client.Get`, parses the `config.yaml` key, and falls back to defaults on NotFound, missing key, invalid YAML, or invalid duration values
+- [x] 1.4 Add tests for YAML parsing: valid full config, partial config, invalid durations, invalid YAML, empty data, missing key
+- [x] 1.5 Add tests for `ConfigMapConfigSource.Load`: ConfigMap exists, ConfigMap not found, ConfigMap without config.yaml key
+
+## 2. Adapter integration
+
+- [x] 2.1 Define `ConfigSource` interface in `internal/adapter` with `Load(ctx context.Context) config.Config`
+- [x] 2.2 Add `ConfigSource` field to `Adapter` struct and accept it in `New()`
+- [x] 2.3 Update `reconcile()` to call `ConfigSource.Load()` at the start and use the returned values for initialDelay and cooldownWindow
+- [x] 2.4 Update `Run()` to detect pollInterval changes and reset the ticker
```
[the remark this answers is not in the document]
Comment (tremes) [reply openspec/changes/external-configmap-config/tasks.md]: This is fixed now.
